src/update-0.8.md

```diff
@@ -0,0 +1,100 @@
+# Updating to 0.8
+
+## Dependencies
+
+Rand crates now require `rustc` version 1.36.0 or later.
+This allowed us to remove some unsafe code and simplify the internal `cfg` logic.
+
+## Core features
+
+[`Rng::gen_range`] now takes a `Range` instead of two numbers.
+This requires replacing `gen_range(a, b)` with `gen_range(a..b)` in code written
+for `rand 0.7`. If `a` or `b` were a reference, explicit dereferencing may now be
+required. Additionally, inclusive ranges are now supported: `gen_range(a, b +
+1)` can be replaced with `gen_range(a..=b)`.
```

Comment:
Include a regex? Replacing `gen_range\(([^,]*),\s*([^)]*)\)` with `gen_range(\1..\2)` does the trick, roughly.

Reply:
Where does the `\1` syntax come from? I think `regex` uses `$1` instead.

Reply:
I just use whatever works with the Kate editor (KDE); `$1` doesn't. According to Wikipedia this is the POSIX standard: https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended

Reply:
I see. The `\i` syntax are backreferences, which are not supported by `regex` (but are by `fancy_regex`). I switched the syntax to `$i` (which I think cannot be used for matching, only for replacing), because this is supported by `regex`.

Reply:
Which `regex` exactly? Oh, the Rust crate? Regexes are much older than Rust.

It appears that `ripgrep` supports the dollar syntax. Perl's and Python's standard regular expressions use backslash for backreferences, as does Emacs...

Reply:
> It appears that ripgrep supports the dollar syntax. Perl's and Python's standard regular expressions use backslash for backreferences, as does Emacs...

I think the main reason for the dollars syntax is that `ripgrep` and the `regex` crate don't support backreferences at all. Therefore, `$1` is not just a different syntax, it is a way to refer to the the first group in the previously matched regular expression without resorting to backreferences.

However, it seems like the dollar syntax is not very common, so I think I'll revert it to the backreference syntax.